js/adapt-contrib-bookmarking.js

```diff
@@ -15,6 +23,33 @@ class Bookmarking extends Backbone.Controller {
     this.listenToOnce(Adapt, 'router:location', this.onAdaptInitialize);
   }
 
+  get globals() {
+    return Adapt.course.get('_globals');
+  }
+
+  get config() {
+    return Adapt.course.get('_bookmarking');
+  }
+
+  get isEnabled() {
+    return Boolean(this.config?._isEnabled);
+  }
+
+  get navigateToId() {
+    const locationConfig = this.config._location || 'previous';
+    const navigateToId = (locationConfig === 'furthest')
+      ? this.furthestIncompleteModel.get('_id')
```

Comment:
If all models are `_isComplete: true` then this line will give a `cannot look up property get on undefined` error. Just need to work that out and decide what happens in `navigateTo`, when set to `furthest` as a consequence.

Reply:
Aye. In our project work, we removed the button in this case. I suppose we would follow our rules and disable the button in this case.

And thinking further, we are disabling these buttons in varied cases. Do you suppose it best to build in some console warnings depending on the situation?